Meme/Meme/MemeTemplatesViewController.swift

```diff
@@ -23,11 +32,18 @@ class MemeTemplatesViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+        self.tableView.register(MemeTemplatesTableViewCell.self, forCellReuseIdentifier: cellID)
+        self.tableView.rowHeight = UITableView.automaticDimension
+        self.tableView.estimatedRowHeight = 250
         self.viewModel.requestTemplates { [weak self] result in
             switch result {
             case .success(let templates):
                 self?.templates = templates
+                DispatchQueue.main.async {
```

Comment:
Perhaps add this as part of the didSet of the templates var

Reply:
So I moved the the DispatchQueue.main.async together with the tableview reload code to the didSet for templates, and I noticed a lag in the loading of the images whenever I scrolled fast. When I changed it back to how it is currently, the loading became much faster. Do you know why moving it to `didSet` is causing the load to slow down?

Reply:
hmm it shouldn't cause a slow down as its just moving the call 🤔 
```swift
var templates: [Template] = [] {
    didSet {
        DispatchQueue.main.async { self.tableView.reloadData() }
    }
}
```
This was what you had?

Reply:
Tried it again after the image caching refactor and seems fine 👍